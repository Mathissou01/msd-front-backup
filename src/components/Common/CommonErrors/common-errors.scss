@use "src/styles/01-settings/settings.colors" as colors;

.c-CommonErrors {
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: 15px 0;
  color: colors.$error;
  font-weight: bold;
}
