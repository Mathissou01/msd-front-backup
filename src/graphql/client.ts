import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache(),
  });
}

export default client;
