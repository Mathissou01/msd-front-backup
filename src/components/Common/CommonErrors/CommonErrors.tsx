import { ApolloError } from "@apollo/client";
import { ReactNode } from "react";
import "./common-errors.scss";

export type TErrorDisplayMode = "replace" | "under" | "toaster";

interface ICommonErrorsProps {
  children: ReactNode;
  errors?: Array<ApolloError | undefined>;
  displayMode?: TErrorDisplayMode;
}

export default function CommonErrors({
  children,
  errors,
  displayMode = "replace",
}: ICommonErrorsProps) {
  /* Static Data */
  const errorTitle = "Erreur";

  /* Local Data */
  const hasErrors = errors && errors?.some((error) => error);

  /* Log Error in console */
  errors?.map((error) => {
    if (error && process.env.NODE_ENV !== "test") {
      process.env.NODE_ENV !== "production"
        ? console.error(error)
        : console.error(error?.message);
    }
  });

  if (hasErrors && displayMode === "toaster") {
    // TODO: implement error handling with "toaster" notifications, or redirects
    return <>{children}</>;
  } else if (hasErrors && displayMode !== "toaster") {
    const errorBlock = (
      <div className="c-CommonErrors">
        {errors?.map((error, index) => {
          if (error && error.message) {
            return (
              <span key={`error_${index}`}>{`${error.name ?? errorTitle}: ${
                error?.message
              }`}</span>
            );
          }
        })}
      </div>
    );

    switch (displayMode) {
      case "replace":
        return errorBlock;
      case "under":
        return (
          <>
            {children}
            {errorBlock}
          </>
        );
    }
  } else {
    return <>{children}</>;
  }
}
