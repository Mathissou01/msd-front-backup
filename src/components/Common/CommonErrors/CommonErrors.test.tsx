import { render } from "@testing-library/react";
import { ApolloError } from "@apollo/client";
import CommonErrors from "./CommonErrors";

const errors = [new ApolloError({ errorMessage: "error message" })];

describe("CommonErrors", () => {
  it("renders", () => {
    const { container } = render(
      <CommonErrors errors={errors} displayMode="under">
        <span>Child Element</span>
      </CommonErrors>,
    );
    expect(container).toMatchSnapshot();
  });
});
