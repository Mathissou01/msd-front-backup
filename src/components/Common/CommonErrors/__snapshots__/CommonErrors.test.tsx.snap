// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonErrors renders 1`] = `
<div>
  <span>
    Child Element
  </span>
  <div
    class="c-CommonErrors"
  >
    <span>
      ApolloError: error message
    </span>
  </div>
</div>
`;
